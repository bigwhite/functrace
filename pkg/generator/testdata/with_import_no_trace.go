package main

import "fmt"

type Foo struct{}

func (Foo) F() {
	D()
}

func A1() {
	B1()
}

func B1() {
	C1()
}

func C1() {
	D()
}

func D() {
}

func main() {
	A1()
	fmt.Println("ok")
}
