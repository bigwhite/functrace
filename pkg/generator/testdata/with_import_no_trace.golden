package main

import (
	"fmt"
	"github.com/pengxuan37/functrace"
)

type Foo struct{}

func (Foo) F() {
	defer functrace.Trace()()
	D()
}

func A1() {
	defer functrace.Trace()()
	B1()
}

func B1() {
	defer functrace.Trace()()
	C1()
}

func C1() {
	defer functrace.Trace()()
	D()
}

func D() {
	defer functrace.Trace()()
}

func main() {
	defer functrace.Trace()()
	A1()
	fmt.Println("ok")
}
